pkg/nsx/services/subnetport/builder.go

```diff
@@ -68,14 +79,14 @@ func (service *SubnetPortService) buildSubnetPort(obj interface{}, nsxSubnetPath
 			TrafficTag:        common.Int64(0),
 			Type_:             String("STATIC"),
 		},
-		Tags:       tags,
+		Tags:       tagsFiltered,
 		Path:       &nsxSubnetPortPath,
 		ParentPath: &nsxSubnetPath,
 	}
 	if appId != "" {
 		nsxSubnetPort.Attachment.AppId = &appId
-		nsxSubnetPort.Attachment.ContextId = &contextID
 	}
+	nsxSubnetPort.Attachment.ContextId = &contextID
```

Comment:
I'm not sure whether the combo `ContextId != nil` and `AppId == nil` can work. Did you test it?

Reply:
I only test subnetport_controller could work as expected as doc describes, I'm not sure whether pod_controller can work since there is no e2e test case.

Reply:
Yes, after hacking, it can work.
Before:
<img width="710" alt="image" src="https://github.com/vmware-tanzu/nsx-operator/assets/4385934/f46efecf-4955-4f84-97fb-3550baeb533d">
After:
<img width="655" alt="image" src="https://github.com/vmware-tanzu/nsx-operator/assets/4385934/cd703c45-8b51-46bc-917a-6d91a242a460">